### PR TITLE
video-search : google

### LINKED ISSUE #320: No option to search videos
With reference to #58 

**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behavior:**
There is no option to search videos in Search engines

**Expected behavior:**
There should be option to search videos

**Open To claim**

### PR DESCRIPTION
**Pull request by umangahuja1**
Fixes #320

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Ongoing : Video feature for other search engines.
- Scope for UI improvement.

### CODE REVIEW

File: app/scrapers/generalized.py [hunk 1]
```diff
@@ -20,12 +21,14 @@ class Scraper:
...
-    def get_page(self, query, startIndex=0):
+    def get_page(self, query, startIndex=0, qtype=''):
         """ Fetch the google search results page
         Returns : Results Page
         """
-        payload = {self.queryKey: query, self.startKey: startIndex}
+        payload = {self.queryKey: query, self.startKey: startIndex,
+                   self.qtype: qtype}
         response = requests.get(self.url, headers=self.headers, params=payload)
+        print(response.url)
```
Comment (abishekvashok) [inline app/scrapers/generalized.py]: Should we do this?

Comment (umangahuja1) [reply app/scrapers/generalized.py]: Well that was during testing. I forgot to remove it :sweat_smile: 
I don't know about live link. Can you tell me more?

Comment (vaibhavsingh97) [reply app/scrapers/generalized.py]: @umangahuja1 Deploy it on heroku and share the link here

**Comment by vaibhavsingh97** [comment]
@umangahuja1 Good Job 🎉 
Now take your time and update your PR with support of other search engines.

File: app/scrapers/generalized.py (hunk 1 above)
Comment (umangahuja1) [reply app/scrapers/generalized.py]: I am not able to deploy. It is asking for card details. Though I have deployed my apps for free earlier.

**Comment by realslimshanky** [comment]
@vaibhavsingh97 we can also do it one search engine at a time.

**Comment by Remorax** [comment]
Can you please share a link to test, or screenshots at least? Thank you 😄

**Comment by umangahuja1** [comment]
I have sent one in issues. If you want I can send another of the query you wish.

**Comment by bhaveshAn** [comment]
We actually want to test it via link.
If the PR is ready for review than please share heroku deployment to get it
reviewed quickly.

**Comment by umangahuja1** [comment]
I don't know why but heroku is asking for card details. It has never happened before. Can anybody brief about it?

**Comment by bhaveshAn** [comment]
Check out in your Heroku dashboard, you might have used 5 free apps.

**Comment by umangahuja1** [comment]
No just 2 of them. And it's due to the mongodb add-on.

**Comment by bhaveshAn** [comment]
So remove the add-on if it restricting you to make more apps.
However, in the recent merge we have disabled mongo. I suppose.

**Comment by umangahuja1** [comment]
If that's the case I can rebase the branch and it will do the job.

**Comment by bhaveshAn** [comment]
Are you able to create the app?
Or tell in which command you are having problems. then Share heroku logs

**Comment by bhaveshAn** [comment]
Also, since now we have divided the Issue. So please write `Addresses #<ISSUE_NO>` instead of `Fixes #<ISSUE_NO>`  in the PR description.

**Comment by umangahuja1** [comment]
Here is the link to the deployed app : https://my-query-server.herokuapp.com/
Only google video search is added.

**Pull request merged**